### PR TITLE
More BUNIT fixes for AIAMap

### PR DESCRIPTION
**Pull request by wtbarnes**
This is an add-on to #4451 to better handle units in AIA images.

Update `DN` to `ct` in the case of normalized and un-normalized counts. 

Also, add `BUNIT` in the case where an image contains the `PIXLUNIT` keyword which is not a FITS standard keyword. However, it seems that this is the keyword that AIA has chosen to denote units in their data products. See http://jsoc.stanford.edu/~jsoc/keywords/AIA/AIA02840_K_AIA-SDO_FITS_Keyword_Document.pdf

### CODE REVIEW

File: sunpy/map/sources/sdo.py
Comment (dstansby) [inline sunpy/map/sources/sdo.py:51]:
```suggestion
        if 'bunit' not in self.meta and 'pixlunit' in self.meta:
```
This guards against a user set 'bunit' being overwritten.

**Review by dstansby — CHANGES_REQUESTED** [review]

**Comment by wtbarnes** [comment]
Should we delete `PIXLUNIT` after updating `BUNIT` as it is a non-standard keyword and potentially conflicts with `BUNIT`?

**Comment by wtbarnes** [comment]
These test failures are real and are the result of the metadata between the saved and in-memory map not being the same because of the addition of the `BUNIT` key.

**Comment by wtbarnes** [comment]
Hmm I take back my previous comment. I can't seem to reproduce this locally.

**Comment by wtbarnes** [comment]
This is failing simply because of the order in which these keys appear. Why are we testing the ordering of the keys?
```
E         -           ('bunit', 'ct'),
E                     ('comment',
E                      'FITS (Flexible Image Transport System) format is defined in '
E                      "'Astronomyand Astrophysics', volume 376, page 359; bibcode: "
E                      '2001A&A...376..359H'),
E                     ('history', ''),
E                     ('keycomments',
E                      {'BITPIX': 'array data type',
E                       'NAXIS': 'number of array dimensions',
E         -             'SIMPLE': 'conforms to FITS standard'})],
E         ?                                                    -
E         +             'SIMPLE': 'conforms to FITS standard'}),
E         +           ('bunit', 'ct')],
E           )

sunpy/map/tests/test_mapbase.py:373: AssertionError
```

**Comment by nabobalis** [comment]
We probably shouldn't be.

**Comment by wtbarnes** [comment]
Yeah I've adjusted the check to be independent of order. I'm not sure if we're doing this elsewhere.

**Comment by dstansby** [comment]
> Should we delete `PIXLUNIT` after updating `BUNIT` as it is a non-standard keyword and potentially conflicts with `BUNIT`?
> 
> I suppose precedent would probably say no as we do not do this in any other `Map` source.

See https://github.com/sunpy/sunpy/issues/4194 for discussion around this. I think what we really should do is keep the original header, and then modify a copy (in which case deleting pixlunit would be fine on the copy). I think this PR is fine for now though.

**Review by dstansby — APPROVED** [review]
But I think it would be worth waiting for the doc build to be fixed elsewhere and passed here before merging.

**Comment by nabobalis** [comment]
Why would the doc build be important here?

**Comment by dstansby** [comment]
🤷‍♂️ not sure, but it has caught bugs that the test suite hasn't caught in the past.

**Comment by wtbarnes** [comment]
I'm quite confused by those doc build failures but I haven't yet tried to reproduce them locally. I don't think they're related to my PR at all, but I've not seen that before and am concerned

**Comment by nabobalis** [comment]
It is fixed by https://github.com/sunpy/sunpy/pull/4466

**Comment by wtbarnes** [comment]
Ok I suspected that was the case. I can wait until that is merged and then rebase to do the docs build check. There's no urgency here.

**Comment by nabobalis** [comment]
It has now been merged in, so the doc build should pass with a rebase.

**Review by dstansby — APPROVED** [review]
If/when tests pass

File: sunpy/map/tests/test_mapbase.py
Comment (nabobalis) [inline sunpy/map/tests/test_mapbase.py:376]: Can we not compare `.items()` directly instead of a loop?

Comment (nabobalis) [reply sunpy/map/tests/test_mapbase.py:376]: Google seems to suggest not.

File: sunpy/map/sources/sdo.py
Comment (nabobalis) [inline sunpy/map/sources/sdo.py:52]:
```suggestion
			# PIXLUNIT is not a FITS standard keyword
            self.meta['bunit'] = self.meta['pixlunit']
```

File: sunpy/map/sources/sdo.py
Comment (nabobalis) [inline sunpy/map/sources/sdo.py:57]:
```suggestion
        # DN and DN/s are not FITS standard units, so convert to counts
```

**Review by nabobalis — APPROVED** [review]

File: sunpy/map/sources/sdo.py
Comment (wtbarnes) [reply sunpy/map/sources/sdo.py:57]: I'd argue that it's really just `DN` that's non-standard and that any composite unit containing DN is then obviously non-standard

File: sunpy/map/tests/test_mapbase.py
Comment (wtbarnes) [reply sunpy/map/tests/test_mapbase.py:376]: I'm not sure. I know this is not the nicest approach. I'm happy to do something different here if there is a better option.

**Pull request merged**
